bigframes/core/indexes/base.py

```diff
@@ -115,6 +115,34 @@ def from_frame(
         index._linked_frame = frame
         return index
 
+    def to_frame(self, index: bool = True, name=None) -> bigframes.dataframe.DataFrame:
+        from bigframes.core.indexes.multi import MultiIndex
+
+        multi = isinstance(self, MultiIndex)
+
+        if multi:
+            columns = [
+                [self.values[j][i] for j in range(len(self.values))]
```

Comment:
self.values calls to_numpy, which downloads everything and not scalable. Maybe take a look at Index.from_frame() and Series.to_frame()

Reply:
fixed for `Index`. per offline conversation with @TrevorBergeron, efficient `MultiIndex` implementation is blocked by bugs in `DataFrame` construction with `MultiIndex`

bug TL;DR

1) Index with array values being created for DataFrame instead of MultiIndex
```
bf_idx = indexes.MultiIndex.from_arrays([["a", "b", "c"], ["d", "e", "f"]])
test = bigframes.dataframe.DataFrame({1: [2, 4, 6], 2: [1, 3, 5]}, index = bf_idx)
print(test)
```
2) block joining for DataFrame creation using MultiIndex and Dict of Series data throws
```
bf_idx = indexes.MultiIndex.from_arrays([["a", "b", "c"], ["d", "e", "f"]])
nlevels = bf_idx.nlevels
columns : list[bigframes.series.Series] = []
 for level in range(nlevels):
       series = self.get_level_values(level).to_series()
       columns.append(series)
data = {i: column for i, column in enumerate(columns)}
result = bigframes.dataframe.DataFrame(
            data, index= bf_idx
        )
```